docs/getting-started.rst

```diff
@@ -0,0 +1,249 @@
+
+.. note::
+   The `Quickstart`_ section provides a working minimal example of code that follows these patterns.
+
+Implementing ThunderbirdComponentResources
+""""""""""""""""""""""""""""""""""""""""""
+
+So you want to develop a new pattern to stamp out? Here's what you'll need to do:
+
+* Determine the best place to put the code. Is there an existing module that fits the bill?
+* Determine the Pulumi type string for it. This goes: ``org:module:class``. The ``org`` should be unique to your
+  organization. For Thunderbird projects, it should be ``tb``. The ``module`` will be the Python submodule you're
+  placing the new class in. The ``class`` is whatever you've called the class.
+* Design the class following these guidelines:
+    * The constructor should always accept, before any other arguments, the following positional options:
+        * ``name``: The internal name of the resource as Pulumi tracks it.
+        * ``project``: The ThunderbirdPulumiProject these resources belong to.
+    * The constructor should always accept the following keyword arguments:
+        * ``opts``: A ``pulumi.ResourceOptions`` object which will get merged into the default set of arguments managed
+          by the project.
+    * The constructor should explicitly define only those arguments that you intend to have default values which differ
+      from the default values the provider will set, or which imply larger patterns (such as ``build_jumphost`` implying
+      other resources, like a security group and its rules, not just an EC2 instance).
+    * The constructor may accept a final ``**kwargs`` argument with arbitrary meaning. Because the nature of a component
+      resource is to compile many other resources into one class, it is not implicitly clear what "everything else"
+      should apply to. If this is implemented, its function should be clearly documented in the class.
+    * The class should extend ``tb_pulumi.ThunderbirdComponentResource``.
+    * The class should call its superconstructor in the following way:
+        * ``super().__init__(typestring, name, project, opts=opts)``
+    * Any resources you create should always be assigned a key in ``self.resources``.
+    * Any resources you create must have the ``parent=self`` ``pulumi.ResourceOption`` set.
+    * At the end of the ``__init__`` function, you must call ``self.finish()``
+
+
+Troubleshooting
+---------------
+
+
+Pythonic problems
+^^^^^^^^^^^^^^^^^
+
+This Pulumi code is developed against Python 3.12 or later. If this is not your default version, you'll need to manage
+your own virtual environment.
+
+Check your default version:
+::
+
+  $ python -V
+  Python 3.12.6
+
+If you need a newer Python, `download and install it <https://www.python.org/downloads/>`_. Then you'll have to set up
+the virtual environment yourself with something like this:
+::
+
+  virtualenv -p /path/to/python3.12 venv
```

Comment:
I'd rather suggest using the currently installed python's venv to install the venv. So `python -m venv venv` (venv being a module and the second venv being the folder.)